tsdb/head_wal.go

```diff
@@ -205,56 +206,12 @@ Outer:
 				if chunks.HeadSeriesRef(h.lastSeriesID.Load()) < walSeries.Ref {
```

Comment:
Yes. I was talking about tombstones and exemplars here. They also need to go inside the processing function because they depend on series record being processed in order. Currently with this change we risk trying to process tombstones or exemplars of a series before it is even created in the memory.

Reply:
If I understand correctly, I think the order remains unchanged since I didn't change anything related to `getOrCreateWithID`.